presto-hive/src/main/java/com/facebook/presto/hive/HivePartitioningHandle.java

```diff
@@ -49,21 +91,37 @@ public int getBucketCount()
     }
 
     @JsonProperty
-    public List<HiveType> getHiveTypes()
+    public Optional<List<HiveType>> getHiveTypes()
     {
         return hiveTypes;
     }
 
+    @JsonProperty
+    public Optional<List<Type>> getTypes()
+    {
+        return types;
+    }
+
     @JsonProperty
     public OptionalInt getMaxCompatibleBucketCount()
     {
         return maxCompatibleBucketCount;
     }
 
+    @JsonProperty
+    public BucketFunctionType getBucketFunctionType()
+    {
+        return bucketFunctionType;
+    }
+
     @Override
     public String toString()
     {
-        return format("buckets=%s, hiveTypes=%s", bucketCount, hiveTypes);
+        return format(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: bucketCount. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 120. Consider formatting this data with &#39;%d&#39; instead.